src/hotspot/share/memory/metaspaceShared.cpp

```diff
@@ -709,9 +709,13 @@ void MetaspaceShared::preload_and_dump_impl(TRAPS) {
   log_info(cds)("Rewriting and linking classes: done");
 
 #if INCLUDE_CDS_JAVA_HEAP
-  if (use_full_module_graph()) {
-    HeapShared::reset_archived_object_states(CHECK);
-  }
+    if (use_full_module_graph()) {
+      HeapShared::reset_archived_object_states(THREAD);
+      if (HAS_PENDING_EXCEPTION) {
```

Comment:
I think this is not necessary. If an exception is thrown, it will be be printed by `MetaspaceShared::preload_and_dump()`

Reply:
Do you mean I should keep the original code as follows?

```
 712   if (use_full_module_graph()) {
 713     HeapShared::reset_archived_object_states(CHECK);
 714   }
```

Reply:
Yes.

Reply:
Oh, you're right. Sorry I missed that :-(